experiment/gerrit-onboarder/onboarder.go

```diff
@@ -0,0 +1,357 @@
+	}
+
+	return newData != string(data), nil
+}
+
+func configToMap(configFile string) (map[string][]string, []string) {
+	configMap := map[string][]string{}
+	orderedKeys := []string{}
+	var curKey string
+	if configFile == "" {
+		return configMap, orderedKeys
+	}
+	for _, line := range strings.Split(configFile, "\n") {
+		if strings.HasPrefix(line, "[") {
+			curKey = line
+			orderedKeys = append(orderedKeys, line)
+		} else {
+			if curList, ok := configMap[curKey]; ok {
+				configMap[curKey] = append(curList, line)
+			} else {
+				configMap[curKey] = []string{line}
+			}
+		}
+	}
+	return configMap, orderedKeys
+}
+
+func mapToConfig(configMap map[string][]string, orderedIDs []string) string {
+	res := ""
+	for _, header := range orderedIDs {
+		res = res + header + "\n"
+		for _, line := range configMap[header] {
+			res = res + line + "\n"
+		}
+	}
+	if res == "" {
+		return res
+	}
+	return strings.TrimSpace(res) + "\n"
+}
+
+func contains(s []string, v string) bool {
+	for _, item := range s {
+		if strings.TrimSpace(item) == strings.TrimSpace(v) {
+			return true
+		}
+	}
+	return false
+}
+
+func getInheritedRepo(configMap map[string][]string) string {
+	if section, ok := configMap["[access]"]; ok {
+		for _, line := range section {
+			if strings.Contains(line, "inheritFrom") {
+				return strings.TrimSpace(strings.Split(line, "=")[1])
+			}
+		}
+	}
+	return ""
+}
+
+func addSection(header string, configMap map[string][]string, configOrder, neededLines []string) (map[string][]string, []string) {
+	if _, ok := configMap[header]; !ok {
+		configMap[header] = []string{}
+		configOrder = append(configOrder, header)
+	}
+	for _, line := range neededLines {
+		configMap[header] = append(configMap[header], "\t"+line)
+	}
+
+	return configMap, configOrder
+}
+
+func labelExists(configMap map[string][]string) bool {
+	_, ok := configMap[LABEL_HEADER]
+	return ok
+}
+
+func lineInRightHeaderFunc(header, line string) func(map[string][]string) bool {
+	return func(configMap map[string][]string) bool {
+		if value, ok := configMap[header]; ok && contains(value, line) {
+			return true
+		}
+		return false
+	}
+}
+
+func labelAccessExistsFunc(groupName string) func(map[string][]string) bool {
+	return func(configMap map[string][]string) bool {
+		for _, value := range configMap {
+			for _, item := range value {
+				if strings.HasPrefix(strings.TrimSpace(item), "label-Verified =") && strings.HasSuffix(strings.TrimSpace(item), fmt.Sprintf("group %s", groupName)) {
+					return true
+				}
+			}
+		}
+		return false
+	}
+}
+
+func verifyInTree(host, cur_branch string, configMap map[string][]string, verify func(map[string][]string) bool) (bool, error) {
+	if verify(configMap) {
+		return true, nil
+	} else if inheritance := getInheritedRepo(configMap); inheritance != "" {
+		parent_branch := cur_branch + "_parent"
+		if err := fetchMetaConfig(host, inheritance, parent_branch); err != nil {
```

Comment:
This is fetching in the same directory, which might not be easy to understand/debug, can we do it in a separate directory?

Reply:
Do you mean clone the repo in a temp dir and create the PR there? I prefer doing it in their own copy of the repo because that way users can make changes to the PR easily if needed. Why would doing it in another dir be easier to debug?

Reply:
IIUC this is fetching a different repo, generally not the best practice in git to clone different repos under the same path. But since `refs/meta/configs` is so different, I think it might be fine. So up to you to decide

Reply:
My original plan was to create a temp dir, clone the parent repo, and fetching from there, but I realized this was a lot easier. Given that we are just reading and not writing to the other repos I don't think we will run into issues. If we do, we can make the change in a future PR.